src/Form/EntityBrowserForm.php

```diff
@@ -140,6 +143,29 @@ public function buildForm(array $form, FormStateInterface $form_state) {
   }
 
   /**
+   * Check if entity browser with selected configuration combination can work.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form status.
+   */
+  protected function isFunctionalForm(FormStateInterface $form_state) {
+    $current_widget_config = $this->entity_browser
+      ->getWidgets()
+      ->get($this->getCurrentWidget($form_state))
+      ->getConfiguration();
+
+    if ($current_widget_config['settings']['auto_select']) {
```

Comment:
This assumes internal configuration of the plugin. From shouldn't assume anything about the internals of the plugins. They should communicate just through functions defined on interfaces or events. Can we refactor this to work through a function?